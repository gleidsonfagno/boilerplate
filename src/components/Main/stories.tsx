// import React from 'react'
// import Main from '.'

// export default {
//   title: 'Main',
//   component: Main
// }

// export const Basic = () => <Main />

import React from 'react'
import { withKnobs, text } from '@storybook/addon-knobs'
import Main from '.'
import { Meta, Story } from '@storybook/react'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
} as Meta

export const Basic: Story = () => (
  <Main
    title={text('Title', 'React AvanÃ§ado')}
    description={text(
      'Description',
      'TypeScript, ReactJS, NextJS e Styled Components'
    )}
  />
)
