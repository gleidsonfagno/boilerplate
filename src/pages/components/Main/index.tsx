import React from 'react'
const Main = () => (
  <main>
    <h1>React avanÃ§ado</h1>
  </main>
)

export default Main
