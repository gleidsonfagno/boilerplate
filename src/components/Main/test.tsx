import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom' // Importação direta do pacote jest-dom
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)
    const headingElement = screen.getByRole('heading', {
      name: /react avançado/i
    })
    expect(headingElement).toBeInTheDocument() // Chamada direta da função

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
