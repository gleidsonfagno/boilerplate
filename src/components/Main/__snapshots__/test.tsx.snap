// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-dzx1ao-0 kqbSjW"
>
  <img
    alt="Imagem de um átomo e React avançado escrito ao lado "
    class="styles__Logo-sc-dzx1ao-1 kupODZ"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-dzx1ao-2 hrVWpn"
  >
    React avançado
     
  </h1>
  <h2
    class="styles__Description-sc-dzx1ao-3 cCzsOU"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tla com código"
    class="styles__Illustration-sc-dzx1ao-4 ffptcN"
    src="/img/hero-illustration.svg"
  />
</main>
`;
